### DISCUSSION TITLE
Rest-times-and-TSMC

### DISCUSSION THREAD

**Discussion by bgyarfas** [sn4k3/UVtools#335] [General]
Awesome run-down on the deficiencies of the "TSMC" feature on the latest chitubox firmware (side note: worst named feature ever). I tried out your gcode suggestion on my Elegoo Saturn running `V4.4.3-Z1-LCDZ` and also checked the settings afterwards with
```
M8093 I1 
M8512 "eeprom_dump.gcode"
```

For the firmware running on my Saturn the factory defaults are:
```
M8006 I5
M8007 I2
```

After running your suggested values, I checked the dumped eeprom and to my surprise saw
```
M8006 I80
M8007 I20
```

My best guess is that the `M8006` and `M8007` commands only accept integer values and the firmware appears to `floor` the value to zero which in turn probably isn't a valid value so it sets it to a hardcoded default. 

So ironically, the settings you were going for inadvertently made the Jerk and Max. Starting Speed _higher_ than the default

By modifying your suggested values to 
```
M8006 I2
M8007 I1
```

then they are reflected correctly in the firmware. 

Not sure if this is a specific case to the Saturn but thought I'd let you know!

**Comment by sn4k3** [comment]
Yes, another user reported that on reddit: https://www.reddit.com/r/resinprinting/comments/qbfq3h/help_with_saturn_using_chitubox_19_needed_angled/

I will update the script to be I1.

BTW can you tell the difference in lifts after apply this mod?

**Reply by bgyarfas** [reply]
If I set the acceleration to 50 (i.e. `M8008 I50`) then I can definitely hear the acceleration difference, I'm not sure about the jerk/max_start_speed yet.  The interesting thing I noticed was if you _don't_ set the `2) height`  or `2) retracts` (or if you have a v3 ctb) then the firmware _forces_ TSMC on you (even if using Chitubox 1.9). There's a distinct speed change at around 2-3mm which corresponds with the following commands in the default gcode
```
;��Z-axis speed and distance��During the Z-axis movement, first slowly rise, then quickly rise, then stay for a while, then quickly drop, and finally slow down
M8028 S3.0;     Unit is mm/s; the speed at which Z rises rapidly (the fast rise distance is equal to the total lift distance minus the slow rise distance)
M8015 P2.0;     Unit is mm/s; Z speed at slow rise
M8070 S3.5;       Unit is mm; Z distance when slowly rising


M8016 P3.5;     Unit is mm/s; the speed of Z fast return (the fast return distance is equal to the total return distance minus the slow return distance)
M8020 S1.2;     Unit is mm/s; Z speed at slow return
M8021 S2.0;     Unit is mm;   Z distance at slow return


;��Z lift distance , the distance = slow rising distance + fast rising distance��
M8070 Z7     ; if motion z is not required, set it to 0
```

**Reply by bgyarfas** [reply]
I just checked and sure enough if you set
```
M8070 S0.0
M8021 S0.0
``` 
then you can disable TSMC

**Reply by sn4k3** [reply]
Nice finding! I never tought that was possible to remove the second lift! I will add that on the template...
But the M8070 states that is the Slow rising? Meaning the first lift and not the second? M8021 states the same for retract (Last retract ie: not TSMC)?

**Edit:** I come down to this lines now:

```
M8006 I1   ; Maximum starting speed mm/s     (60mm/min) [Should be 0.7 but firmware constrains anything lower than 1]
M8007 I1   ; Jerk mm/s                       (60mm/min) [Should be 0.4 but firmware constrains anything lower than 1]
M8008 I400 ; Acceleration mm/s2
M8013 I4   ; Maximum speed of Z motion mm/s  (240mm/min)
M8016 I1   ; Second home speed: Decreasing the second home speed improve the positioning accuracy of the home limit
M8070 S0.0 ; Disable the slow rising         (Disable in firmware TSMC)
M8021 S0.0 ; Disable the slow retract        (Disable in firmware TSMC)
;M8015 T4  ; Slow maximum limit rise speed   (240mm/min)
;M8016 T4  ; Fast maximum limit fall speed   (240mm/min)
M8015 P4.0 ; Slow maximum limit rise speed   (240mm/min)
M8028 S4.0 ; Fast maximum limit rise speed   (240mm/min)
M8020 S4.0 ; Slow maximum limit fall speed   (240mm/min)
M8016 P4.0 ; Fast maximum limit fall speed   (240mm/min)


M8500      ; Save configuration permanently
```

**Reply by bgyarfas** [reply]
> But the M8070 states that is the Slow rising? Meaning the first lift and not the second? 
I think that's right. 

I'm pretty confident since the ctb file I used to test whether or not I saw a "second" speed had the following entries set to 0 
```
2) Bottom Lift Height
2) Lift Height
2) Bottom Retract Height
2) Retract Height
```
And I would see a change in the movement speed if I changed the other Retract/Lift Speeds (all without a second speed during a lift/retract)

It's a shame this version of the motherboard isn't using a motor driver like the TMC5041, then it could have proper acceleration/deceleration profiles... Currently, I've set my acceleration to `I10` and it has a real nice acceleration/deceleration for the lift/retract but I wish on the opposite end (i.e. when the platform is high) the deceleration was more abrupt since it's just wasting time slowing down for no reason.

**Reply by sn4k3** [reply]
OK i have test it, and it works, i also put I10 accel as 400 produce same high speeds.

> it has a real nice acceleration/deceleration for the lift/retract but I wish on the opposite end

Thats true, but you asking for to much 🤣
It's a shame we can't configure all parameters we like, thats the big down side of closed firmwares, we are stuck with what is given and old machines will never see a update either.

I updated file again

**Reply by bgyarfas** [reply]
Awesome. Thanks again for working on UVTools, I'm really enjoying it 🥇
